### PR TITLE
prevent crash when reaction emoji is removed while ReactionsScreen is…

### PR DESCRIPTION
**Pull request by carlisgg**
#### Summary
Original issue shows an index out of range error and according to screenshot it seems to be happening in the post list screen. After test and code review there is a guard already added in post_list some months ago that should be covering this crash committed months after issue report:

```
    const scrollToIndex = useCallback((index: number, animated = true, applyOffset = true) => {
        if (index < 0 || !listRef?.current) {
            return;
        }
       ....
    }
```

After reviewing other scrollToIndex similar code locations, there is a **crash also happening when showing emoji reactions list** bottom sheet for a post.
This fix covers and tests also that particular scenario and makes sure all other scrollToIndex calls are guarded for out of range indexing.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-66558

#### Checklist
- [x] Added or updated unit tests (required for all new features)
- [ ] Have run E2E tests by adding label `E2E/Run` (or `E2E/Run-iOS` / `E2E/Run-Android` for platform-specific runs).

#### Device Information
This PR was tested on: Android 16 & iOS 26.1 emulators

#### Release Note
```release-note
fixes crash while showing emoji reaction list
```

### CODE REVIEW

**Review by enahum — APPROVED** [review]
Great!! I had the crash the other day and was going to look into it.

Thanks

File: app/screens/reactions/reactions.tsx [hunk 1]
```diff
@@ -32,7 +32,7 @@ function getSortedReactions(reactions: ReactionModel[] | undefined, prevSortedRe
 
 const Reactions = ({initialEmoji, location, reactions}: Props) => {
     const [sortedReactions, setSortedReactions] = useState(Array.from(new Set(reactions?.map((r) => getEmojiFirstAlias(r.emojiName)))));
-    const [index, setIndex] = useState(sortedReactions.indexOf(initialEmoji));
+    const [index, setIndex] = useState(() => Math.max(0, sortedReactions.indexOf(initialEmoji)));
```
[the remark this answers is not in the document]
Comment (carlisgg) [reply app/screens/reactions/reactions.tsx:36]: this is fine since it covers an edge case

File: app/screens/reactions/emoji_bar/index.tsx [hunk 2]
```diff
@@ -36,6 +36,10 @@ const EmojiBar = ({emojiSelected, reactionsByName, setIndex, sortedReactions}: P
     const listRef = useRef<FlatList<string>>(null);
 
     const scrollToIndex = (index: number, animated = false) => {
+        if (index < 0 || !listRef?.current) {
+            return;
+        }
+
```
[the remark this answers is not in the document]
Comment (carlisgg) [reply app/screens/reactions/emoji_bar/index.tsx:42]: real scenarios fail when index is not found but tend to provide proper indexes otherwise

File: app/screens/reactions/reactions.tsx [hunk 3]
```diff
@@ -54,7 +54,7 @@ const Reactions = ({initialEmoji, location, reactions}: Props) => {
 
     const renderContent = useCallback(() => {
         const emojiAlias = sortedReactions[index];
-        if (!reactionsByName) {
+        if (!emojiAlias || !reactionsByName) {
             return null;
         }
```
[the remark this answers is not in the document]
Comment (carlisgg) [reply app/screens/reactions/reactions.tsx:59]: current check is enough

**Comment by enahum** [comment]
@amyblais I think we may want to include this one in the dot release for 2.41 that fixes the android crashes, but this fixes a crash in both platforms.

**Comment by amyblais** [comment]
> I think we may want to include this one in the dot release for 2.41 that fixes the android crashes, but this fixes a crash in both platforms.

@enahum Thank you, Mobile v2.41.2 has been submitted to the app store reviews earlier today.
